src/app/pages/cinemahall/cinemahall.component.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, OnInit } from '@angular/core';
+import { Observable } from 'rxjs';
+import { mergeRequest } from 'src/app/servises/hint/request';
+import { concatResponses } from 'src/app/servises/hint/response';
+import { MediaResult } from 'src/app/servises/model/response-data';
+import { PopularMediaService } from 'src/app/servises/popular-media.service';
+import { TrendingMediaService } from 'src/app/servises/trending-media.service';
+import { UpcomingMediaService } from 'src/app/servises/upcoming-media.service';
+
+interface Sections {
+  title: string,
+  content: MediaResult[]
+}
+
+type sectionType = 'popular' | 'trending' | 'upcoming'
+
+@Component({
+  selector: 'mov-cinemahall',
+  templateUrl: './cinemahall.component.html',
+  styleUrls: ['./cinemahall.component.scss']
+})
+export class CinemahallComponent implements OnInit {
+  constructor(
+    private readonly apiPopularMedia: PopularMediaService,
+    private readonly apiTrendingMedia: TrendingMediaService,
+    private readonly apiUpcomingMedia: UpcomingMediaService
+  ){
+
+    mergeRequest<MediaResult[]>([{type: 'movie'}, {type: 'tv'}], this.apiPopularMedia.requestMediaByType.bind(this.apiPopularMedia))
+      .subscribe((data:MediaResult[][]) => {
+        this.MediaData.popular.content = concatResponses(data)
+      }
+    )
+    mergeRequest<MediaResult[]>([{type: 'movie'}, {type: 'tv'}], this.apiTrendingMedia.requestMediaByType.bind(this.apiTrendingMedia))
+      .subscribe((data:MediaResult[][]) => {
+        this.MediaData.trending.content = concatResponses(data)
+      }
+    )
+    mergeRequest<MediaResult[]>([{type: 'movie'}, {type: 'tv'}], this.apiUpcomingMedia.requestMediaByType.bind(this.apiUpcomingMedia))
+      .subscribe((data:MediaResult[][]) => {
+        this.MediaData.upcoming.content = concatResponses(data)
+      }
+    )
+  }
```

Comment:
Avoid direct api calls from components, use data services instead, eg.: 
`popularMedia.getMedia()`

Reply:
mergeRequest - is an utile function which helps with multiple request.

I am not sure with this solution

Reply:
Let's leave it for now as cinema hall component will fully changed.